src/hooks/useClickOutside/useClickOutside.test.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { useRef } from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import { useOnClickOutside } from '.';
+
+describe('[hooks] useClickOutside', () => {
+  it('should call handler when clicked outside', () => {
+    const mockFn = jest.fn();
+
+    const TestComponent = () => {
+      const ref = useRef(null);
+      useOnClickOutside(ref, mockFn);
+
+      return (
+        <div>
+          <div ref={ref} data-testid="inside"></div>
+          <div data-testid="outside"></div>
+        </div>
+      );
+    };
+
+    const { getByTestId } = render(<TestComponent />);
```

Comment:
https://testing-library.com/docs/dom-testing-library/api-queries/#queries

getByTestId is the last one you'd wanna use in a real world testing scenario. For this test in particular I believe it's fine but try to keep in mind that you should be testing what users are actually seeing.

Reply:
It's quite difficult to serve something users are actually seeing, it's a hook so it does not really display something.